classes/class-admin-notices.php

```diff
@@ -0,0 +1,1033 @@
+			</p>
+		</div>
+		<?php
+		$this->wc_regen_script();
+	}
+
+	/**
+	 * Loads the inline script to dismiss the WC regen notice.
+	 */
+	public function wc_regen_script() {
+		?>
+		<script>
+			jQuery(document).on('click', '#ewww-image-optimizer-wc-regen .notice-dismiss', function() {
+				var ewww_dismiss_wc_regen_data = {
+					action: 'ewww_dismiss_wc_regen',
+					_wpnonce: <?php echo wp_json_encode( wp_create_nonce( 'ewww-image-optimizer-notice' ) ); ?>,
+				};
+				jQuery.post(ajaxurl, ewww_dismiss_wc_regen_data, function(response) {
+					if (response) {
+						console.log(response);
+					}
+				});
+			});
+		</script>
+		<?php
+	}
+
+	/**
+	 * Lets the user know LR Sync has regenerated thumbnails and that they need to take action.
+	 */
+	public function lightroom_sync_notice() {
+		if ( ! current_user_can( apply_filters( 'ewww_image_optimizer_admin_permissions', '' ) ) ) {
+			return;
+		}
+		if ( ! \class_exists( '\Meow_WPLR_Sync_Core' ) || ! $this->get_option( 'ewww_image_optimizer_lr_sync' ) ) {
+			return;
+		}
+		?>
+		<div id='ewww-image-optimizer-lr-sync' class='notice notice-info is-dismissible'>
+			<p>
+				<?php \esc_html_e( 'EWWW Image Optimizer has detected a WP/LR Sync process. To optimize new thumbnails, you may run the Bulk Optimizer from the Media menu. This notice may be dismissed after the Sync process is complete.', 'ewww-image-optimizer' ); ?>
+			</p>
+		</div>
+		<?php
+		$this->lightroom_sync_script();
+	}
+
+	/**
+	 * Loads the inline script to dismiss the LR sync notice.
+	 */
+	public function lightroom_sync_script() {
+		?>
+		<script>
+			jQuery(document).on('click', '#ewww-image-optimizer-lr-sync .notice-dismiss', function() {
+				var ewww_dismiss_lr_sync_data = {
+					action: 'ewww_dismiss_lr_sync',
+					_wpnonce: <?php echo wp_json_encode( wp_create_nonce( 'ewww-image-optimizer-notice' ) ); ?>,
+				};
+				jQuery.post(ajaxurl, ewww_dismiss_lr_sync_data, function(response) {
+					if (response) {
+						console.log(response);
+					}
+				});
+			});
+		</script>
+		<?php
+	}
+
+	/**
+	 * Requires the removal of Animated Gif Resize plugin.
+	 */
+	public function agr_notice() {
+		if ( ! current_user_can( apply_filters( 'ewww_image_optimizer_admin_permissions', '' ) ) ) {
+			return;
+		}
+		if ( ! \class_exists( '\Bbpp_Animated_Gif' ) ) {
+			return;
+		}
+		?>
+		<div id="ewww-image-optimizer-notice-agr" class="notice notice-warning">
+			<p>
+				<?php \esc_html_e( 'GIF animations are preserved by EWWW Image Optimizer automatically. Please remove the Animated GIF Resize plugin.', 'ewww-image-optimizer' ); ?>
+			</p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Checks for exec() and availability of local optimizers, then displays an error if needed.
+	 *
+	 * @param string $quiet Optional. Use 'quiet' to suppress output.
+	 */
+	public function utils_notice( $quiet = null ) {
+		$this->debug_message( '<b>' . __METHOD__ . '()</b>' );
+		if ( ewwwio()->cloud_mode ) {
+			return;
+		}
```

Comment:
Once upon a time, I suspect the plugin would bypass the rest of this if the hosting was known to block exec, or if the OS was unsupported. Currently, two notices are shown in such cases, so we need to add these checks:
`if ( ewwwio()->local->hosting_requires_api() ) {`
and
`if ( ! ewwwio()->local->os_supported() ) {`